### ISSUE TITLE
Error when publishing an ASP.NET Core app (preview4)

### ISSUE THREAD

**Issue by nalla** [dotnet/cli#5373]
## Steps to reproduce
None. I ran `dotnet publish` on a private repo. Will try add then when I can reproduce with a public repo.

## Expected  behavior
The application gets published.

## Actual behavior
dotnet.exe fails with MSB error complaining that there is a missmatch between `SourceFiles` and `DestinationFiles`.

## Environment data
dotnet.exe preview4 (MSBuild based)

## Bug Hunt
While on the hunt I found the following piece of code in

     sdk/1.0.0-preview4-004233/Sdks/Microsoft.NET.Sdk/build/Microsoft.NET.Publish.targets

on line `100`

    <Copy SourceFiles = "@(_ResolvedFileToPublishPreserveNewest)"       
          DestinationFiles="$(PublishDir)%(_ResolvedFileToPublishPreserveNewest.RelativePath)"                                                  
          OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"            
          Retries="$(CopyRetryCount)"                                   
          RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"        
          UseHardlinksIfPossible="$(CreateHardLinksForPublishFilesIfPossible)"                                                                  
          UseSymboliclinksIfPossible="$(CreateSymbolicLinksForPublishFilesIfPossible)">

Changing the `SourceFiles` parameter to

    %(_ResolvedFileToPublishPreserveNewest.FullPath)

 helped and the application was published.

If you look more closely on the parameters then those familiar with MSBuild will notice that an array is always the source and a batched relative path is the destination. This is OK as long as the array only contains one file.

So I hope someone from the team can make use of the information here and might have a look at that MSBuild target more closely.

**Comment by piotrpMSFT** [comment]
@nalla 1.0.0-preview4-004233 is a fairly old build. Please upgrade to the latest [rc3] from the home page of this repo. If you still see issues, please open a new issue.

**Comment by JonDouglas** [comment]
@piotrpMSFT I also get a similar error when using VS 2017 RC(Which I believe invokes a different version of tooling). I've also tried downloading the most recent RC4 builds and can't get past the error defined above when publishing through VS 2017. Is there a way to work around this?

**Comment by JonDouglas** [comment]
I can confirm that > `1.0.0-preview4-004233` solves this issue locally, unfortunately Azure is on that build and fails to publish:

```
D:\Program Files (x86)\dotnet\sdk\1.0.0-preview4-004233\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Publish.targets(100,5): error MSB3094: "DestinationFiles" refers to 1 item(s), and "SourceFiles" refers to 2 item(s). They must have the same number of items.
```

**Comment by piotrpMSFT** [comment]
@JonDouglas I'm glad the fix is already in CLI.

I haven't played with the Azure Web sites much. Do you have the ability to run your own build script? You can consider having a script download 1.0.0-preview4-004233 and manually kicking off a build [there are user-local ZIPs available for each release]. This would get you unblocked as new bits roll out to Azure!

**Comment by JonDouglas** [comment]
One of the PMs on that team had let me know that although that situation is possible, it is not documented. However they had noted they update every couple weeks or so. Thank you for your diligence. It is greatly appreciated.
